076-moving-data-between-two-jbod-disks-using-cruise-control.md

```diff
@@ -0,0 +1,89 @@
+
+### Implementation
+
+To implement this feature, we will be adding a new mode to the `KafkaRebalanceMode` class.
+* `remove-disks`: It moves replicas from a specified disk to other disks of the same broker. It always uses intra-broker re-balancing.
+You can use this mode by changing the `spec.mode` to `remove-disks` in the `KafkaRebalance` resource.
+
+A `KafkaRebalance` custom resource would look like this.
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaRebalance
+metadata:
+  name: my-rebalance
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  # setting the mode as `remove-disks` to move data between the JBOD disks
+  mode: remove-disks
+  # providing the list of brokers, and the corresponding volumes from which you want to move the replicas
+  moveReplicasOffVolumes:
+    - brokerId: 0
+      volumeIds: [1, 2]
+    - brokerId: 2
+      volumeIds: [1]
+# ...
+```
+
+### Flow
+
+- The user should be using the `Kafka` resource with JBOD configured. Make sure that you have more than one disk configured on the brokers.
+- When the Kafka cluster is ready, the user creates a `KafkaRebalance` custom resource with the `spec.mode` field as `remove-disks` and provides list of the brokers, and the corresponding volumes from which you want to move the replicas in the `spec.moveReplicasOffVolumes` field.
+- The `KafkaRebalanceAssemblyOperator` interacts with Cruise Control via the `/remove_disks` endpoint to generate an optimization proposal (by using the dryrun feature).
+- You can use `strimzi.io/rebalance-auto-approval:true` annotation on the `KafkaRebalance` resource for auto-approval of proposal. In case you want to do it manually you can do it by applying the `strimzi.io/rebalance=approve` annotation on it.
+- The `KafkaRebalanceAssemblyOperator` interacts with Cruise Control via the `/remove_disks` endpoint to perform the actual rebalancing.
+
+Note: The optimization proposal will not show the load before optimization, it will only show the load after optimization. This is because in upstream Cruise Control, we don't have the verbose tag enabled with the `remove_disks` endpoint.
+
+### Other Scenarios
+
+- In case the user is not using JBOD storage and tries to generate the optimization proposal, the `KafkaRebalance` resource will move to `NotReady` state prompting invalid log dirs provided for the broker.
+- If you are using JBOD with single disk configured on the brokers, in that case `KafkaRebalance` will move to `NotReady` state prompting that you don't have enough log dirs to move the replicas for that broker.
+- If the disk capacity has exceeded for the broker, in that case `KafkaRebalance` will move to `NotReady` prompting that enough capacity is not remaining to move replicas for that broker.
+- This feature works fine with KafkaNodePools. 
```

Comment:
Do we need this sentence? What does it add/mean? We have everything now moving to node pools (mostly because KRaft) so I would expect that any new features will work fine with node pools as explicit.

Reply:
There was earlier a commet by @scholzj, if it will work fine with nodepools and KRaft, therefore I thought it would be good to mention about all of them. I agree, any new feature should work with nodepool but just for clarity maybe we can still have this pointer.